### ISSUE TITLE
rails 7 rc1 method_missing: undefined method `assets'

### ISSUE THREAD

**Issue by lrjbrual** [rails/rails#43793]
### Steps to reproduce
Update gem from rails 7.0.0.alpha2 to gem "rails", "~> 7.0.0.rc1"
bundle update / bundle install
bin/dev

development.rb with  config.assets.quiet = true

### Expected behavior
Run properly for rails 7 rc1

### Actual behavior
```
20:36:08 web.1  | /Users/ryanjeffbrual/.rvm/gems/ruby-3.0.1/gems/railties-7.0.0.rc1/lib/rails/railtie/configuration.rb:96:in `method_missing': undefined method `assets' for #<Rails::Application::Configuration:0x00007fe8a2410b30 @root=#<Pathname:/Users/ryanjeffbrual/gorails/doctor_app>, @generators=#<Rails::Configuration::Generators:0x00007fe8d219e098 @aliases={}, @options={:rails=>{:orm=>:active_record, :test_framework=>:test_unit, :integration_tool=>:test_unit, :system_tests=>:test_unit}, :active_record=>{:migration=>true, :timestamps=>true}, :test_unit=>{:fixture=>true, :fixture_replacement=>nil}}, @fallbacks={}, @templates=[], @colorize_logging=true, @api_only=false, @hidden_namespaces=[], @after_generate_callbacks=[]>, @middleware=#<Rails::Configuration::MiddlewareStackProxy:0x00007fe8a24105b8 @operations=[], @delete_operations=[]>, @javascript_path="javascript", @encoding=#<Encoding:UTF-8>, @allow_concurrency=nil, @consider_all_requests_local=true, @filter_parameters=[], @filter_redirect=[], @helpers_paths=[], @hosts=[".localhost", #<IPAddr: IPv4:0.0.0.0/0.0.0.0>, #<IPAddr: IPv6:0000:0000:0000:0000:0000:0000:0000:0000/0000:0000:0000:0000:0000:0000:0000:0000>], @host_authorization={}, @public_file_server=#<ActiveSupport::OrderedOptions {:enabled=>true, :index_name=>"index"}>, @force_ssl=false, @ssl_options={:hsts=>{:subdomains=>true}}, @session_store=nil, @time_zone="UTC", @beginning_of_week=:monday, @log_level=:debug, @cache_store=:null_store, @railties_order=[:all], @relative_url_root=nil, @reload_classes_only_on_change=true, @file_watcher=ActiveSupport::FileUpdateChecker, @exceptions_app=nil, @autoflush_log=true, @log_formatter=#<ActiveSupport::Logger::SimpleFormatter:0x00007fe8d249f008 @datetime_format=nil>, @eager_load=false, @secret_key_base=nil, @api_only=false, @debug_exception_response_format=nil, @x=#<Rails::Application::Configuration::Custom:0x00007fe8d249ef68 @configurations={}>, @enable_dependency_loading=false, @read_encrypted_secrets=false, @content_security_policy=nil, @content_security_policy_report_only=false, @content_security_policy_nonce_generator=nil, @content_security_policy_nonce_directives=nil, @require_master_key=false, @loaded_config_version=7.0, @credentials=#<ActiveSupport::OrderedOptions {:content_path=>#<Pathname:/Users/ryanjeffbrual/gorails/doctor_app/config/credentials.yml.enc>, :key_path=>#<Pathname:/Users/ryanjeffbrual/gorails/doctor_app/config/master.key>}>, @disable_sandbox=false, @add_autoload_paths_to_load_path=true, @permissions_policy=nil, @rake_eager_load=false, @server_timing=false, @paths=#<Rails::Paths::Root:0x00007fe8d24a7910 @path=#<Pathname:/Users/ryanjeffbrual/gorails/doctor_app>, @root={"app"=>#<Rails::Paths::Path:0x00007fe8d24a7730 @paths=["app"], @current="app", @root=#<Rails::Paths::Root:0x00007fe8d24a7910 ...>, @glob="{*,*/concerns}", @exclude=["assets", "javascript"], @autoload_once=false, @eager_load=true, @autoload=false, @load_path=false>, "app/assets"=>#<Rails::Paths::Path:0x00007fe8d24a75c8 @paths=["app/assets"], @current="app/assets", @root=#<Rails::Paths::Root:0x00007fe8d24a7910 ...>, @glob="*", @exclude=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "app/controllers"=>#<Rails::Paths::Path:0x00007fe8d24a7500 @paths=["app/controllers"], @current="app/controllers", @root=#<Rails::Paths::Root:0x00007fe8d24a7910 ...>, @glob=nil, @exclude=nil, @autoload_once=false, @eager_load=true, @autoload=false, @load_path=false>, "app/channels"=>#<Rails::Paths::Path:0x00007fe8d24a7460 @paths=["app/channels"], @current="app/channels", @root=#<Rails::Paths::Root:0x00007fe8d24a7910 ...>, @glob=nil, @exclude=nil, @autoload_once=false, @eager_load=true, @autoload=false, @load_path=false>, "app/helpers"=>#<Rails::Paths::Path:0x00007fe8d24a73c0 @paths=["app/helpers"], @current="app/helpers", @root=#<Rails::Paths::Root:0x00007fe8d24a7910 ...>, @glob=nil, @exclude=nil, @autoload_once=false, @eager_load=true, @autoload=false, @load_path=false>, "app/models"=>#<Rails::Paths::Path:0x00007fe8d24a7320 @paths=["app/models"], @current="app/models", @root=#<Rails::Paths::Root:0x00007fe8d24a7910 ...>, @glob=nil, @exclude=nil, @autoload_once=false, @eager_load=true, @autoload=false, @load_path=false>, "app/mailers"=>#<Rails::Paths::Path:0x00007fe8d24a7280 @paths=["app/mailers"], @current="app/mailers", @root=#<Rails::Paths::Root:0x00007fe8d24a7910 ...>, @glob=nil, @exclude=nil, @autoload_once=false, @eager_load=true, @autoload=false, @load_path=false>, "app/views"=>#<Rails::Paths::Path:0x00007fe8d24a71e0 @paths=["app/views"], @current="app/views", @root=#<Rails::Paths::Root:0x00007fe8d24a7910 ...>, @glob=nil, @exclude=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "lib"=>#<Rails::Paths::Path:0x00007fe8d24a7140 @paths=["lib"], @current="lib", @root=#<Rails::Paths::Root:0x00007fe8d24a7910 ...>, @glob=nil, @exclude=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=true>, "lib/assets"=>#<Rails::Paths::Path:0x00007fe8d24a7078 @paths=["lib/assets"], @current="lib/assets", @root=#<Rails::Paths::Root:0x00007fe8d24a7910 ...>, @glob="*", @exclude=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "lib/tasks"=>#<Rails::Paths::Path:0x00007fe8d24a6fd8 @paths=["lib/tasks"], @current="lib/tasks", @root=#<Rails::Paths::Root:0x00007fe8d24a7910 ...>, @glob="**/*.rake", @exclude=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "config"=>#<Rails::Paths::Path:0x00007fe8d24a6f60 @paths=["config"], @current="config", @root=#<Rails::Paths::Root:0x00007fe8d24a7910 ...>, @glob=nil, @exclude=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "config/environments"=>#<Rails::Paths::Path:0x00007fe8d24a6e48 @paths=["config/environments"], @current="config/environments", @root=#<Rails::Paths::Root:0x00007fe8d24a7910 ...>, @glob="development.rb", @exclude=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "config/initializers"=>#<Rails::Paths::Path:0x00007fe8d24a6da8 @paths=["config/initializers"], @current="config/initializers", @root=#<Rails::Paths::Root:0x00007fe8d24a7910 ...>, @glob="**/*.rb", @exclude=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "config/locales"=>#<Rails::Paths::Path:0x00007fe8d24a6d08 @paths=["config/locales"], @current="config/locales", @root=#<Rails::Paths::Root:0x00007fe8d24a7910 ...>, @glob="**/*.{rb,yml}", @exclude=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "config/routes.rb"=>#<Rails::Paths::Path:0x00007fe8d24a6c90 @paths=["config/routes.rb"], @current="config/routes.rb", @root=#<Rails::Paths::Root:0x00007fe8d24a7910 ...>, @glob=nil, @exclude=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "config/routes"=>#<Rails::Paths::Path:0x00007fe8d24a6bf0 @paths=["config/routes"], @current="config/routes", @root=#<Rails::Paths::Root:0x00007fe8d24a7910 ...>, @glob="**/*.rb", @exclude=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "db"=>#<Rails::Paths::Path:0x00007fe8d24a6b78 @paths=["db"], @current="db", @root=#<Rails::Paths::Root:0x00007fe8d24a7910 ...>, @glob=nil, @exclude=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "db/migrate"=>#<Rails::Paths::Path:0x00007fe8d24a6b00 @paths=["db/migrate"], @current="db/migrate", @root=#<Rails::Paths::Root:0x00007fe8d24a7910 ...>, @glob=nil, @exclude=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "db/seeds.rb"=>#<Rails::Paths::Path:0x00007fe8d24a6a88 @paths=["db/seeds.rb"], @current="db/seeds.rb", @root=#<Rails::Paths::Root:0x00007fe8d24a7910 ...>, @glob=nil, @exclude=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "vendor"=>#<Rails::Paths::Path:0x00007fe8d24a69e8 @paths=["vendor"], @current="vendor", @root=#<Rails::Paths::Root:0x00007fe8d24a7910 ...>, @glob=nil, @exclude=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=true>, "vendor/assets"=>#<Rails::Paths::Path:0x00007fe8d24a6948 @paths=["vendor/assets"], @current="vendor/assets", @root=#<Rails::Paths::Root:0x00007fe8d24a7910 ...>, @glob="*", @exclude=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "config/database"=>#<Rails::Paths::Path:0x00007fe8d24a68a8 @paths=["config/database.yml"], @current="config/database", @root=#<Rails::Paths::Root:0x00007fe8d24a7910 ...>, @glob=nil, @exclude=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "config/secrets"=>#<Rails::Paths::Path:0x00007fe8d24a6808 @paths=["config"], @current="config/secrets", @root=#<Rails::Paths::Root:0x00007fe8d24a7910 ...>, @glob="secrets.yml{,.enc}", @exclude=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "config/environment"=>#<Rails::Paths::Path:0x00007fe8d24a6768 @paths=["config/environment.rb"], @current="config/environment", @root=#<Rails::Paths::Root:0x00007fe8d24a7910 ...>, @glob=nil, @exclude=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "lib/templates"=>#<Rails::Paths::Path:0x00007fe8d24a66f0 @paths=["lib/templates"], @current="lib/templates", @root=#<Rails::Paths::Root:0x00007fe8d24a7910 ...>, @glob=nil, @exclude=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "log"=>#<Rails::Paths::Path:0x00007fe8d24a6600 @paths=["log/development.log"], @current="log", @root=#<Rails::Paths::Root:0x00007fe8d24a7910 ...>, @glob=nil, @exclude=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "public"=>#<Rails::Paths::Path:0x00007fe8d24a6588 @paths=["public"], @current="public", @root=#<Rails::Paths::Root:0x00007fe8d24a7910 ...>, @glob=nil, @exclude=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "public/javascripts"=>#<Rails::Paths::Path:0x00007fe8d24a6510 @paths=["public/javascripts"], @current="public/javascripts", @root=#<Rails::Paths::Root:0x00007fe8d24a7910 ...>, @glob=nil, @exclude=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "public/stylesheets"=>#<Rails::Paths::Path:0x00007fe8d24a6498 @paths=["public/stylesheets"], @current="public/stylesheets", @root=#<Rails::Paths::Root:0x00007fe8d24a7910 ...>, @glob=nil, @exclude=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>, "tmp"=>#<Rails::Paths::Path:0x00007fe8d24a6420 @paths=["tmp"], @current="tmp", @root=#<Rails::Paths::Root:0x00007fe8d24a7910 ...>, @glob=nil, @exclude=nil, @autoload_once=false, @eager_load=false, @autoload=false, @load_path=false>}>, @autoload_paths=["/Users/ryanjeffbrual/gorails/doctor_app/app/madmin/resources", "/Users/ryanjeffbrual/gorails/doctor_app/app/madmin/fields"], @cache_classes=false> (NoMethodError)
20:36:08 web.1  | Did you mean?  asset_host
20:36:08 web.1  | 	from /Users/ryanjeffbrual/gorails/doctor_app/config/environments/development.rb:58:in `block in <main>'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/.rvm/gems/ruby-3.0.1/gems/railties-7.0.0.rc1/lib/rails/railtie.rb:251:in `instance_eval'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/.rvm/gems/ruby-3.0.1/gems/railties-7.0.0.rc1/lib/rails/railtie.rb:251:in `configure'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/gorails/doctor_app/config/environments/development.rb:3:in `<main>'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/.rvm/gems/ruby-3.0.1/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `require'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/.rvm/gems/ruby-3.0.1/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:23:in `block in require_with_bootsnap_lfi'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/.rvm/gems/ruby-3.0.1/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/loaded_features_index.rb:100:in `register'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/.rvm/gems/ruby-3.0.1/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:22:in `require_with_bootsnap_lfi'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/.rvm/gems/ruby-3.0.1/gems/bootsnap-1.9.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:31:in `require'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/.rvm/gems/ruby-3.0.1/gems/zeitwerk-2.5.1/lib/zeitwerk/kernel.rb:35:in `require'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/.rvm/gems/ruby-3.0.1/gems/railties-7.0.0.rc1/lib/rails/engine.rb:562:in `block (2 levels) in <class:Engine>'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/.rvm/gems/ruby-3.0.1/gems/railties-7.0.0.rc1/lib/rails/engine.rb:561:in `each'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/.rvm/gems/ruby-3.0.1/gems/railties-7.0.0.rc1/lib/rails/engine.rb:561:in `block in <class:Engine>'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/.rvm/gems/ruby-3.0.1/gems/railties-7.0.0.rc1/lib/rails/initializable.rb:32:in `instance_exec'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/.rvm/gems/ruby-3.0.1/gems/railties-7.0.0.rc1/lib/rails/initializable.rb:32:in `run'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/.rvm/gems/ruby-3.0.1/gems/railties-7.0.0.rc1/lib/rails/initializable.rb:61:in `block in run_initializers'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/.rvm/rubies/ruby-3.0.1/lib/ruby/3.0.0/tsort.rb:228:in `block in tsort_each'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/.rvm/rubies/ruby-3.0.1/lib/ruby/3.0.0/tsort.rb:350:in `block (2 levels) in each_strongly_connected_component'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/.rvm/rubies/ruby-3.0.1/lib/ruby/3.0.0/tsort.rb:422:in `block (2 levels) in each_strongly_connected_component_from'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/.rvm/rubies/ruby-3.0.1/lib/ruby/3.0.0/tsort.rb:431:in `each_strongly_connected_component_from'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/.rvm/rubies/ruby-3.0.1/lib/ruby/3.0.0/tsort.rb:421:in `block in each_strongly_connected_component_from'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/.rvm/gems/ruby-3.0.1/gems/railties-7.0.0.rc1/lib/rails/initializable.rb:50:in `each'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/.rvm/gems/ruby-3.0.1/gems/railties-7.0.0.rc1/lib/rails/initializable.rb:50:in `tsort_each_child'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/.rvm/rubies/ruby-3.0.1/lib/ruby/3.0.0/tsort.rb:415:in `call'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/.rvm/rubies/ruby-3.0.1/lib/ruby/3.0.0/tsort.rb:415:in `each_strongly_connected_component_from'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/.rvm/rubies/ruby-3.0.1/lib/ruby/3.0.0/tsort.rb:349:in `block in each_strongly_connected_component'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/.rvm/rubies/ruby-3.0.1/lib/ruby/3.0.0/tsort.rb:347:in `each'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/.rvm/rubies/ruby-3.0.1/lib/ruby/3.0.0/tsort.rb:347:in `call'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/.rvm/rubies/ruby-3.0.1/lib/ruby/3.0.0/tsort.rb:347:in `each_strongly_connected_component'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/.rvm/rubies/ruby-3.0.1/lib/ruby/3.0.0/tsort.rb:226:in `tsort_each'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/.rvm/rubies/ruby-3.0.1/lib/ruby/3.0.0/tsort.rb:205:in `tsort_each'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/.rvm/gems/ruby-3.0.1/gems/railties-7.0.0.rc1/lib/rails/initializable.rb:60:in `run_initializers'
20:36:08 web.1  | 	from /Users/ryanjeffbrual/.rvm/gems/ruby-3.0.1/gems/railties-7.0.0.rc1/lib/rails/application.rb:369:in `initialize!'
```

### System configuration
**rails 7.0.0.alpha2**:

**Ruby version 3.0.1**:

**Comment by rafaelfranca** [comment]
Thank you for the issue. You need to add `sprockets-rails` in your Gemfile since it is not a dependency of Rails anymore.

**Comment by lrjbrual** [comment]
@rafaelfranca, sorry I don't understand but I'm not using sprocket. my rails version is from rails 7 alpha2 then i switch to rc1. I'm using gem "jsbundling-rails", "~> 0.1.0", gem "cssbundling-rails" if not mistaken we are not using sprocket from that? 

thanks

**Comment by rafaelfranca** [comment]
In that case you only need to remove `config.assets` calls from your application.

**Comment by lrjbrual** [comment]
Thank you @rafaelfranca. I still have the same issue when I remove from assets.rb. 

what I did is I try to create a new app using rails 7rc1 and I did not get any error seems working fine.

and compare the two apps but the only difference I have is from 7 alpha that I upgraded to 7 rc1 is the config.assets that calling the node_modules the rest are the same. 

not sure where I can look further.

**Comment by rafaelfranca** [comment]
Did you remove all references to `config.assets` from your application? It seems you have one in `config/environments/development.rb` as you can see in the backtrace.

**Comment by dwightwatson** [comment]
It looks like the default `Gemfile` now includes `"sprockets-rails", ">= 3.4.1"` which wasn't required previously. This also caught us in the move from beta2 to rc1.

**Comment by lrjbrual** [comment]
@rafaelfranca I already remove(although I do the comment), and in development.rb, assets.rb config.assets are remove but still the same error. here's link for reference:

[https://gist.github.com/lrjbrual/408509bad3d9c3867fe70b7ef0d5e717](https://gist.github.com/lrjbrual/408509bad3d9c3867fe70b7ef0d5e717)

maybe this can help also to other that will have the same issue. until now I did not resolve yet. 

@dwightwatson I don't have sprockets-rails as per validation in our gemfile lock. you can find also the info at my gist above link.

I will continue later to revalidate and debug. but if you guys have any other idea let me know. thank you

**Comment by salex** [comment]
I'm also having a similar problem with a rails 7.0.0alph2 app that was using importmaps

I added 2 gems (rails_slim view components) to a bare alpha2 demo app that only had a demo controller testing some tailwind stuff.

The `bundle update`  then installed rc1. Trying to start the server I got.

```
SteveAir:tw_list_disk $ rails s          
=> Booting Puma
=> Rails 7.0.0.rc1 application starting in development 
=> Run `bin/rails server --help` for more startup options
Exiting
/Users/salex/.rbenv/versions/3.0.1/lib/ruby/gems/3.0.0/gems/railties-7.0.0.rc1/lib/rails/railtie/configuration.rb:96:in 
`method_missing': undefined method `assets' for 
#<Rails::Application::Configuration:0x000000012055e3c8 
@root=#<Pathname:/Users/salex/Work/rails7/tw_list_disk>, @generators=#<Rails::Configuration::Generators:0x000000013289a548 @aliases={}, 
@options={:rails=>{:orm=>:active_record, :test_framework=>:test_unit, :integration_tool=>:test_unit, :system_tests=>:test_unit, :template_engine=>:slim}, :active_record=>{:migration=>true, :timestamps=>true}, :test_unit=>{:fixture=>true, :fixture_replacement=>nil}}, @fallbacks={}, @templates=[], @colorize_logging=true, @api_only=false, @hidden_namespaces=[], @a

.... 

Did you mean?  asset_host

```
I did an rails app:update

```
SteveAir:tw_list_disk $ bin/rails app:update
   identical  config/boot.rb
       exist  config
   identical  config/application.rb
   identical  config/environment.rb
       exist  config/environments
    conflict  config/environments/development.rb
Overwrite /Users/salex/Work/rails7/tw_list_disk/config/environments/development.rb? (enter "h" for help) [Ynaqdhm] y
       force  config/environments/development.rb
    conflict  config/environments/production.rb
Overwrite /Users/salex/Work/rails7/tw_list_disk/config/environments/production.rb? (enter "h" for help) [Ynaqdhm] y
       force  config/environments/production.rb
    conflict  config/environments/test.rb
Overwrite /Users/salex/Work/rails7/tw_list_disk/config/environments/test.rb? (enter "h" for help) [Ynaqdhm] y
       force  config/environments/test.rb
       exist  config/initializers
   identical  config/initializers/assets.rb
    conflict  config/initializers/content_security_policy.rb
Overwrite /Users/salex/Work/rails7/tw_list_disk/config/initializers/content_security_policy.rb? (enter "h" for help) [Ynaqdhm] y
       force  config/initializers/content_security_policy.rb
      create  config/initializers/cors.rb
   identical  config/initializers/filter_parameter_logging.rb
   identical  config/initializers/inflections.rb
      create  config/initializers/new_framework_defaults_7_0.rb
   identical  config/initializers/permissions_policy.rb
      remove  config/initializers/cors.rb
       exist  bin
   identical  bin/rails
   identical  bin/rake
   identical  bin/setup
       rails  active_storage:update
rails aborted!
NoMethodError: undefined method `assets' for #<Rails::Application::Configuration:0x000000013ab50fa0 @root=#<Pathname:/Users/salex/Work/rails7/tw_list_disk>, @generators=#<Rails::Configuration::Generators:0x0000000149e503b8 @aliases={}, @options={:rails=>{:orm=>:active_record, :test_framework=>:test_unit, :integration_tool=>:test_unit, :system_tests=>:test_unit, :template_engine=>:slim}, :active_record=>{:migration=>true, :timestamps=>true}, :t
```

I also created a new demo app (rc1) copied my demo controller, views and routes and it worked fine

I even copied my rc1 config folder and replaced the original alpa2 config folder and it still won't work.

Seems to me (and I don't know much!) the problem is in `railties-7.0.0.rc1`

**Comment by rafaelfranca** [comment]
Your gist still show you have a `config.assets` call in `/Users/ryanjeffbrual/gorails/doctor_app/config/initializers/assets.rb:4` as the backtrace also says. Please read the backtrace, it will tell you where you have the wrong method call. 

if you want to call `config.assets` you need to add `sprockets-rails` to the Gemfile. If you don't want to use it, you need to remove all the `config.assets`.

**Comment by salex** [comment]
It's now there and is now working!

I guess 7.0.0.alpha2 used a different scheme with import maps, since it was not here in the alpha version of my app

**Comment by lrjbrual** [comment]
@rafaelfranca thanks for the confirmation again. I leave out the config.assets to assets.rb because I found when I created new app using rails 7 rc1, I got `Rails.application.config.assets.version = "1.0"` so I did not remove that. but if I will remove the Rails.application.config.assets.version = "1.0", the app will work, however I have an issue with the application.css and javascript/application.js

```
16:55:16 web.1  | ActionController::RoutingError (No route matches [GET] "/stylesheets/application.css"):
16:55:16 web.1  | ActionController::RoutingError (No route matches [GET] "/javascripts/application.js"):
```
which I did understand the issues that that there is no application.css found, however I'm using 
```
gem "jsbundling-rails", "~> 0.1.0"
gem "cssbundling-rails"
```

so I have application.bootstrap.scss the same setup for what I did in rails 7 rc1 from yesterday. 

sorry I got confuse, the assets work now. and if I use the sprockets-rails then my setup will be change because I'm building the .css and .js to the builds assets folder.


@salex how did you resolve it? 

Thanks in advance.

**Comment by lrjbrual** [comment]
It seems that the alpha2 is using the sprockets-rails? but then in the future if this is remove in 7.0rc1 fully and the gem sprockets-rails are not maintain for instance, then it will cause re update the app. but how we can avoid to use this? 
what I'm trying to understand is when I created a rails 7 rc1 I have the same setup assets and using 

```
gem "jsbundling-rails", "~> 0.1.0"
gem "cssbundling-rails"
```
with the Rails.application.config.assets.version = "1.0" in assets.rb 

the same how I bundle the assets to /assets/builds this is seems work it can compile and it can use the bootstrap 5 layouts.

however for the upgraded one which is my project from alpha2 to rc1 is not working, @rafaelfranca you are correct I need to remove the config.assets but the issue is it cannot see the application.css and application.js then I need to dig in how I can add the assets for my app, probably I cannot resolve in given time and it is only can resolve via sprocket then it seems that I'm back again to rails 6 for the frontend but not sure about it, maybe I'm wrong. 

But can we avoid to use sprocket since this is already using rails 7 alpha2? 

Thanks

**Comment by salex** [comment]
@lrjbrual 
Again I was using the importmaps option.  I've used esbuild  and I'am trying to decided what way I'll go when it try to upgrade my apps.

I generated an new rc1 app with `rails new esrc1 -j esbuild --css=tailwind` and it contains `gem "sprockets-rails", ">= 3.4.1"`  and `Rails.application.config.assets.version = "1.0"`

I'm also a little confused myself and it is probably something is missing from going form alpha2 to rc1 that is being missed.
I assume you are using `bin/dev` to start your app. At least I thing that's what does the build process.
